base/filesystem.jl

```diff
@@ -44,10 +44,14 @@ import .Base:
     skip, stat, unsafe_read, unsafe_write, write, transcode, uv_error,
     rawhandle, OS_HANDLE, INVALID_OS_HANDLE, windowserror
 
+import .Base.RefValue
+
 if Sys.iswindows()
     import .Base: cwstring
 end
 
+const MAX_PATH = Sys.iswindows() ? 260 : 1024 # max buffer size including null terminator
```

Comment:
Should we perhaps give this a more accurate name? The real `MAX_PATH` on NT is 32768, on Win32 is 260, on bsd sockets is 108, on linux is 4096, etc.

Since we have logic to resize as needed, perhaps we should call this `AVG_PATH` and define it to always be some moderate sized value (like 108 or 260)?

Reply:
Good point. I'm not sure on which value  makes the most sense on average. I think the current numbers seem quite reasonable on average for non-Windows systems (we could go with `512`)?
For windows it seems reasonable however to just use `260`. 

Renaming the constant makes sense I believe.